dlls/nodes.cpp

```diff
@@ -2386,42 +2387,47 @@ int CGraph::FLoadGraph( const char *szMapName )
 	pMemFile = aMemFile = LOAD_FILE_FOR_ME( szFilename, &length );
 
 	if( !aMemFile )
-	{
 		return FALSE;
-	}
-	else
-	{
-		// Read the graph version number
-		//
-		length -= sizeof(int);
-		if( length < 0 )
-			goto ShortFile;
-		memcpy( &iVersion, pMemFile, sizeof(int) );
-		pMemFile += sizeof(int);
 
-		if( iVersion != GRAPH_VERSION )
-		{
-			// This file was written by a different build of the dll!
-			//
-			ALERT( at_aiconsole, "**ERROR** Graph version is %d, expected %d\n", iVersion, GRAPH_VERSION );
-			goto ShortFile;
-		}
+	// Read the graph version number
+	//
+	length -= sizeof(int);
+	if( length < 0 )
+		goto ShortFile;
+	iVersion = *(int *) pMemFile;
+	pMemFile += sizeof(int);
 
+	if( iVersion == GRAPH_VERSION || iVersion == 16 )
```

Comment:
I suggest you to make defines:
```
#define GRAPH_VERSION_32 16
#define GRAPH_VERSION_64 (GRAPH_VERSION * 10)
```
And check them depending on platform target, instead of using magic values like 160 or 16.
Or, which is better, create our own nodegraph format without dependence on platform `void*`, `string_t`, `long` sizes, with compatibility mode for GRAPH_VERSION 16(or drop it forever, nodes can be recreated if needed).

Reply:
Good point about the magic numbers, I did it like that to keep it platform-agnostic. Like, I personally would prefer:
`#define _GRAPH_VERSION_LEGACY	(16)
#ifdef XASH_64BIT
#define	_GRAPH_VERSION			(16 * 10)
#else
#define	_GRAPH_VERSION			(16)
#endif
`